src/main/java/org/apache/commons/compress/archivers/zip/ZipFile.java

```diff
@@ -899,13 +921,17 @@ public Enumeration<ZipArchiveEntry> getEntriesInPhysicalOrder() {
      * @since 1.6
      */
     public Iterable<ZipArchiveEntry> getEntriesInPhysicalOrder(final String name) {
-        ZipArchiveEntry[] entriesOfThatName = ZipArchiveEntry.EMPTY_ARRAY;
-        final LinkedList<ZipArchiveEntry> linkedList = nameMap.get(name);
-        if (linkedList != null) {
-            entriesOfThatName = linkedList.toArray(entriesOfThatName);
-            Arrays.sort(entriesOfThatName, offsetComparator);
+        if (duplicateNameMap != null) {
+            List<ZipArchiveEntry> list = duplicateNameMap.get(name);
+            if (list != null) {
+                ZipArchiveEntry[] entriesOfThatName = list.toArray(ZipArchiveEntry.EMPTY_ARRAY);
```

Comment:
Instead of: 

1. allocate array
2. sort array
3. allocate list
 
would it be better to change the type from `List` to `LinkedList` (which is what is used in the instance variable decl and:

1. `clone()` the list
2. `Collections.sort()` the cloned list

?

Reply:
> Instead of:
> 
> 1. allocate array
> 2. sort array
> 3. allocate list
> 
> would it be better to change the type from `List` to `LinkedList` (which is what is used in the instance variable decl and:
> 
> 1. `clone()` the list
> 2. `Collections.sort()` the cloned list
> 
> ?

I think it might be cleaner to keep it as it is now.

The return type of the `clone` method of `ArrayList`/`LinkedList` is `Object`, which means that after calling it, a cast is required and `SuppressWarnings` annotation is required to suppress unchecked warnings.